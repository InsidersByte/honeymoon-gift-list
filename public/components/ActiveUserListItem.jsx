/* @flow */

import React, { Component } from 'react';
import { Avatar, ListItem, IconButton } from 'material-ui';
import Person from 'material-ui/svg-icons/social/person';
import Delete from 'material-ui/svg-icons/action/delete';

type PropsType = {
    user: {
        id: string,
        name: string,
        email: string,
    },
    loggedInUser: {
        email: string,
    },
    onDelete: Function,
};

export default class ActiveUserListItem extends Component {
    props: PropsType;

    onDelete = () => {
        this.props.onDelete(this.props.user);
    };

    render() {
        const { user: { id, name, email }, loggedInUser } = this.props;
        const deletable = email !== loggedInUser.email;

        if (deletable) {
            return (
                <ListItem
                    key={id}
                    leftAvatar={<Avatar icon={<Person />} />}
                    rightIconButton={<IconButton onClick={this.onDelete}><Delete /></IconButton>}
                    primaryText={name}
                    secondaryText={email}
                />
            );
        }

        return (
            <ListItem
                key={id}
                leftAvatar={<Avatar icon={<Person />} />}
                primaryText={name}
                secondaryText={email}
            />
        );
    }
}
