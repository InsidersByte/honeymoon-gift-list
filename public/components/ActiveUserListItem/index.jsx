/* @flow */

import React from 'react';
import { Avatar, ListItem, IconButton } from 'material-ui';
import Person from 'material-ui/svg-icons/social/person';
import Delete from 'material-ui/svg-icons/action/delete';
import type { UserType, AuthUser } from '../../types';

type PropsType = {
  user: UserType,
  loggedInUser: AuthUser,
  onDelete: (user: UserType) => void,
};

const ActiveUserListItem = ({ user, loggedInUser, onDelete }: PropsType) => {
  const { id, name, email } = user;
  const handleDelete = () => onDelete(user);
  const deletable = email !== loggedInUser.email;

  return (
    <ListItem
      key={id}
      leftAvatar={<Avatar icon={<Person />} />}
      rightIconButton={deletable ? <IconButton onClick={handleDelete}><Delete /></IconButton> : undefined}
      primaryText={name}
      secondaryText={email}
    />
  );
};

export default ActiveUserListItem;
