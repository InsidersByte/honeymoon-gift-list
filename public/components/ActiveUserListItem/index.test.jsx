import React from 'react';
import { shallow, mount } from 'enzyme';
import { IconButton, MuiThemeProvider } from 'material-ui';
import { getMuiTheme } from 'material-ui/styles';
import injectTapEventPlugin from 'react-tap-event-plugin';
import ActiveUserListItem from './';

injectTapEventPlugin();

describe('ActiveUserListItem', () => {
    it('should render deletable correctly', () => {
        const user = { id: 1, email: 'email@email.com', name: 'person' };
        const loggedInUser = { email: 'anotheremail@email.com' };
        const onDelete = jest.fn();

        const wrapper = shallow(
            <ActiveUserListItem user={user} loggedInUser={loggedInUser} onDelete={onDelete} />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(onDelete).not.toHaveBeenCalled();
    });

    it('should handle delete correctly', () => {
        const user = { id: 1, email: 'email@email.com', name: 'person' };
        const loggedInUser = { email: 'anotheremail@email.com' };
        const onDelete = jest.fn();

        const wrapper = mount(
            <MuiThemeProvider muiTheme={getMuiTheme()}>
                <ActiveUserListItem user={user} loggedInUser={loggedInUser} onDelete={onDelete} />
            </MuiThemeProvider>,
        );

        expect(wrapper.find(IconButton).length).toBe(1);
        wrapper.find(IconButton).simulate('click');

        expect(onDelete).toBeCalledWith(user);
    });

    it('should render not deletable correctly', () => {
        const user = { id: 1, email: 'email@email.com', name: 'person' };
        const loggedInUser = { email: 'email@email.com' };
        const onDelete = jest.fn();

        const wrapper = shallow(
            <ActiveUserListItem user={user} loggedInUser={loggedInUser} onDelete={onDelete} />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(onDelete).not.toHaveBeenCalled();
    });
});
