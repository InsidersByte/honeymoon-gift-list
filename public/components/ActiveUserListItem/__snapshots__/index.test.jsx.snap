// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveUserListItem should render deletable correctly 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  insetChildren={false}
  leftAvatar={
    <Avatar
      icon={<SocialPerson />}
      size={40}
    />
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="person"
  primaryTogglesNestedList={false}
  rightIconButton={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      onClick={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <ActionDelete />
    </IconButton>
  }
  secondaryText="email@email.com"
  secondaryTextLines={1}
/>
`;

exports[`ActiveUserListItem should render not deletable correctly 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  insetChildren={false}
  leftAvatar={
    <Avatar
      icon={<SocialPerson />}
      size={40}
    />
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="person"
  primaryTogglesNestedList={false}
  secondaryText="email@email.com"
  secondaryTextLines={1}
/>
`;
